lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3Context.java

```diff
@@ -30,15 +30,27 @@
 import static io.trino.filesystem.s3.S3FileSystemConstants.EXTRA_CREDENTIALS_SESSION_TOKEN_PROPERTY;
 import static java.util.Objects.requireNonNull;
 
-record S3Context(int partSize, boolean requesterPays, S3SseType sseType, String sseKmsKeyId, Optional<AwsCredentialsProvider> credentialsProviderOverride, ObjectCannedAcl cannedAcl)
+record S3Context(
+        int partSize,
+        boolean requesterPays,
+        S3SseType sseType,
+        String sseKmsKeyId,
+        S3SseCustomerKey sseCustomerKey,
+        Optional<AwsCredentialsProvider> credentialsProviderOverride,
+        ObjectCannedAcl cannedAcl)
 {
     private static final int MIN_PART_SIZE = 5 * 1024 * 1024; // S3 requirement
 
     public S3Context
     {
         checkArgument(partSize >= MIN_PART_SIZE, "partSize must be at least %s bytes", MIN_PART_SIZE);
         requireNonNull(sseType, "sseType is null");
-        checkArgument((sseType != S3SseType.KMS) || (sseKmsKeyId != null), "sseKmsKeyId is null for SSE-KMS");
+        if (sseType == S3SseType.KMS) {
+            checkArgument(sseKmsKeyId != null, "sseKmsKeyId is null for SSE-KMS");
```

Comment:
requireNonNull ?

Reply:
imho, IllegalArgumentException is better suited than NullPointerException in this case 
unless we have policy to use just rnn, I will rewrite it